src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -158,7 +166,19 @@ class AuthScreens extends React.Component {
 
         // Fetch some data we need on initialization
         NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.NVP_PRIORITY_MODE, 'default');
-        NameValuePair.get(CONST.NVP.PREFERRED_LOCALE, ONYXKEYS.NVP_PREFERRED_LOCALE, 'en');
+
+        API.Get({
+            returnValueList: 'nameValuePairs',
+            nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        }).then((response) => {
+            const preferredLocale = response.nameValuePairs.preferredLocale || CONST.DEFAULT_LOCALE;
```

Comment:
👋  @MonilBhavsar there's a pretty weird bug related to this one that @kevinksullivan spotted locally. Unsure if anything super bad is happening as a result, but thought it would be worth mentioning that `response.nameValuePairs` can be an empty array if the nvp we are specifying doesn't exist. 

Here's a response for an nvp that I did have:

![2021-09-15_16-45-22](https://user-images.githubusercontent.com/32969087/133541349-b724b270-7723-4f02-a56d-186b8dc1699f.png)

Followed by one that does not exist

![2021-09-15_16-45-22](https://user-images.githubusercontent.com/32969087/133541386-266535cc-1ecb-4a9d-8ffa-ce0cd1368e23.png)

Basically, not safe to assume this is an object and we should maybe update to use `lodashGet()` for now?

Reply:
👋 Looking into it

Reply:
So this will still default to `CONST.DEFAULT_LOCALE` but it's just safer to use `lodashGet(response.nameValuePairs.preferredLocale)` here right?

Reply:
>So this will still default to CONST.DEFAULT_LOCALE

Actually, yeah I think it should long as `response.nameValuePairs` is an object or array (so what I'm saying above likely is not the reason why this is breaking).

>it's just safer to use lodashGet(response.nameValuePairs.preferredLocale) here right

Yeah, especially when accessing a nested property + we can't guarantee some parent property like `nameValuePairs` exists in the `response` and is also the type we expect it to be. Maybe the real error was because nameValuePairs is `undefined` 🤔  

I'm not sure tbh just saw this line was hit in the JS console and guessed it might have something to do with the array returning instead of an object.

Reply:
Yea, totally fair. I'll look out for this type of pattern in the future (accessing from an object that may not exist), and for now I think we should indeed change this to use `lodashGet()`.